shell/server/grain-server.js

```diff
@@ -335,6 +335,9 @@ Meteor.methods({
       if (grain) {
         if (grain.userId === this.userId) {
           Grains.update({ _id: grainId, userId: this.userId }, { $set: { title: newTitle } });
+
+          // Denormalize new title out to all sharing tokens.
+          ApiTokens.update({ grainId: grainId }, { $set: { "owner.user.upstreamTitle": newTitle } });
```

Comment:
Probably you only want to do this for grains where `"owner.user": { $exists: true }`